File: manual/code/test_suites/python/python/test_tenacity_retry_if_result.py
import unittest
from function_tenacity_retry_if_exception_type import RetryFunction
from tenacity import RetryError

class TestMightFailFunction(unittest.TestCase):

    def setUp(self):
        self.func = RetryFunction()

    def test_success_on_first_try(self):
        result = self.func.might_fail_function(6)
        self.assertEqual(result, 6)

    def test_failure_after_max_attempts(self):
        with self.assertRaises(RetryError) as context:
            self.func.might_fail_function(2, should_raise=True)
        self.assertTrue("ValueError" in str(context.exception))

    def test_retry_on_result(self):
        with self.assertRaises(RetryError) as context:
            self.func.might_fail_function(2, should_raise=True)
        self.assertTrue("ValueError" in str(context.exception))

    def test_no_retry_on_result(self):
        result = self.func.might_fail_function(5)
        self.assertEqual(result, 5)

    def test_retry_with_custom_logic(self):
        with self.assertRaises(RetryError) as context:
            self.func.might_fail_function(3, should_raise=True)
        self.assertTrue("ValueError" in str(context.exception))

if __name__ == '__main__':
    unittest.main()

File: data/java/tests/test_tenacity_retry_if_result.java
package com.example;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_tenacity_retry_if_result {
    private function_tenacity_retry_if_result testInstance;

    @BeforeEach
    void setUp() {
        testInstance = new function_tenacity_retry_if_result();
    }

    @Test
    void test_success_on_first_try() {
        int result = testInstance.might_fail_function(6, false);
        assertEquals(6, result);
    }

    @Test
    void test_failure_after_max_attempts() {
        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
            testInstance.might_fail_function(2, true);
        });
        assertTrue(exception.getMessage().contains("Value is too small"));
    }

    @Test
    void test_retry_on_result() {
        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
            testInstance.might_fail_function(2, true);
        });
        assertTrue(exception.getMessage().contains("Value is too small"));
    }

    @Test
    void test_no_retry_on_result() {
        int result = testInstance.might_fail_function(5, false);
        assertEquals(5, result);
    }

    @Test
    void test_retry_with_custom_logic() {
        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
            testInstance.might_fail_function(3, true);
        });
        assertTrue(exception.getMessage().contains("Value is too small"));
    }
}